packages/scalecube-microservice/tests/integration/Errors/remoteServiceCall.spec.ts

```diff
@@ -0,0 +1,148 @@
+    applyPostMessagePolyfill();
+    applyMessageChannelPolyfill();
+  }
+
+  const greetingServiceDefinition = {
+    serviceName: 'GreetingService',
+    methods: {
+      hello: {
+        asyncModel: ASYNC_MODEL_TYPES.REQUEST_RESPONSE,
+      },
+      greet$: {
+        asyncModel: ASYNC_MODEL_TYPES.REQUEST_STREAM,
+      },
+      incorrectAsyncModel: {
+        asyncModel: ASYNC_MODEL_TYPES.REQUEST_RESPONSE,
+      },
+      incorrectAsyncModel$: {
+        asyncModel: ASYNC_MODEL_TYPES.REQUEST_STREAM,
+      },
+    },
+  };
+  const microserviceWithServices = Microservices.create({
+    services: [
+      {
+        definition: greetingServiceDefinition,
+        reference: {
+          hello: () => Promise.reject(new Error(errorMessage)),
+          greet$: () =>
+            new Observable((obs) => {
+              obs.error(new Error(errorMessage));
+            }),
+          incorrectAsyncModel: () => of({ emptyMessage }),
+          incorrectAsyncModel$: () => Promise.resolve({ emptyMessage }),
+        },
+      },
+    ],
+  });
+
+  const microServiceWithoutServices = Microservices.create({});
+
+  describe.each([
+    // ################# LocalCall #################
+    {
+      sender: microserviceWithServices,
+      receiverServiceDefinition: greetingServiceDefinition,
+      isRemote: false,
+    },
+    // ################# RemoteCall ################
+    {
+      sender: microServiceWithoutServices,
+      receiverServiceDefinition: greetingServiceDefinition,
+      isRemote: true,
+    },
+  ])(
+    `
+        Given  'sender' and a 'receiverServiceDefinition' (receiver)
+               sender (microservice)       | receiverServiceDefinition (receiver)    |
+               microserviceWithServices    | greetingServiceDefinition               | # LocalCall
+               microserviceWithoutServices | greetingServiceDefinition               | # RemoteCallCall
+
+        `,
+    (connect) => {
+      const { sender, receiverServiceDefinition, isRemote } = connect;
+
+      // @ts-ignore
+      if (isRemote && global.isNodeEvn) {
+        return; // TODO: RFC - remoteCall nodejs
+      }
+
+      test(`
+    Scenario: service reject
+    Given     proxy and remoteService
+    When      remoteService reject
+    Then      proxy receive the error message
+
+  `, () => {
+        expect.assertions(1);
+        const proxy = sender.createProxy({ serviceDefinition: receiverServiceDefinition });
+
+        return expect(proxy.hello('Me')).rejects.toMatchObject(new Error(errorMessage));
+      });
+
+      test(`
+    Scenario: service emit error
+    Given     proxy and remoteService
+    When      remoteService emit error
+    Then      proxy receive the error message
+  `, (done) => {
+        expect.assertions(1);
+
+        const proxy = sender.createProxy({ serviceDefinition: receiverServiceDefinition });
+        proxy
+          .greet$(['Me'])
+          .pipe(
+            catchError((error: any) => {
```

Comment:
better not to do it in **catchError**, but do it in error callback itself (the second argument of **subscribe** method)